src/main/java/org/duckstudy/model/car/Cars.java

```diff
@@ -0,0 +1,73 @@
+package org.duckstudy.model.car;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.duckstudy.model.generator.Generator;
+import org.duckstudy.view.OutputView;
+
+public class Cars {
+
+    private static final int INITIAL_POSITION = 0;
+
+    private final List<Car> cars;
+
+    public Cars(List<Car> cars) {
+        this.cars = List.copyOf(cars);
+    }
+
+    public Cars(String[] carNames, Generator generator) {
+        this.cars = createCars(carNames, generator);
```

Comment:
생성자는 보통 안정적인 객체를 보장하는 주생성자 밑에 여러 부생성자를 두는것이 좋아요. 현재는 15~17의 생성자가 된다면,
```suggestion
        this(createCars(carNames, generator));
```
가 될 수 있겠네요.

Reply:
그렇군요 ! createCars가 멤버 메서드여서 this 전에 호출이 불가능하니 this 안에 createCars 메서드 내용을 넣어서 수정하겠습니다 !!
```java
public Cars(String[] carNames, RandomValueGenerator randomValueGenerator, OutputView outputView) {
      this(Arrays.stream(carNames)
              .map(carName -> new Car(carName, randomValueGenerator, outputView))
              .collect(Collectors.toList()));
  }
```